### DISCUSSION TITLE
Time error in log

### DISCUSSION THREAD

**Discussion by kirmesfreak** [dani-garcia/vaultwarden#2399] [Q&A]
I have set up vaultwarden in my docker environment, running nginx as the reverse proxy and fail2ban directly on the OS.

fail2ban displays after setting up the jail and starting well

> 2022-03-27 17:10:11,739 fail2ban.filter         [3023]: WARNING [bitwarden] Simulate NOW in operation since found time has too large deviation None ~ 1648393811.7393708 +/- 60

> 2022-03-27 17:10:11,740 fail2ban.filter         [3023]: WARNING [bitwarden] Please check jail has possibly a timezone issue. Line with odd timestamp: {"log":"[2022-03-27 17:10:11.259][request][INFO] POST /identity/connect/token\r\n","stream":"stdout","time":"2022-03-27T15:10:11.26344437Z"}

Vaultwarden-log.joson 
>{"log":"[2022-03-27 16:47:19.155][start][INFO] Rocket has launched from http://0.0.0.0:80\r\n","stream":"stdout","time":"2022-03-27T14:47:19.157380025Z"}

The date of the docker itself is correct. I also already loaded TZ and /etc/localtime
![Bildschirmfoto 2022-03-27 um 17 44 54](https://user-images.githubusercontent.com/102476499/160289862-fed4f2b6-f328-4f6a-9c38-6c03791b6456.png)

Any ideas how I can change the time in the log so that fail2ban can work?

**Comment by BlackDex** [comment]
First i would check the diagnostics page: `https://your.domain.tld/admin/diagnostics` and check what the diagnostics there say regarding the date/time.

**Reply by kirmesfreak** [reply]
Local and UTC time is showed the right way, I'd say. That means, the date in the log is generally in UTC?
![Bildschirmfoto 2022-03-27 um 19 59 08](https://user-images.githubusercontent.com/102476499/160294377-d9cb8b5f-1772-4525-b6e6-224c1d1d5212.png)

So fail2ban is then more or less the issue?

**Reply by BlackDex** [reply]
Looks like it indeed.
You can check the logs your self or course.
If the current time checks out in the logs it could be fail2ban has a wrong timezones or it can't read the time correctly from the logs

**Reply by kirmesfreak** [reply]
You're right. Anyway, found the issue, Thx :)

**Comment by xkpx64** [comment]
I got proxmox server with correct timedate ( Tue Aug 22 04:42:47 PM EEST 2023 )
On top of it as VM i run OpnSense with chrony server ( on port 123 )
and container of VaultWarden for AlpineLinux( again with correct time and date - and connected to ntp server via ntpsec )
Please help to resolve.
![Untitled](https://github.com/dani-garcia/vaultwarden/assets/5773378/2ff111b1-5707-42c2-acbd-007ed9cbc8b0)

**Reply by stefan0xC** [reply]
If you look at the **Browser** time you can see that your local devices' clock seems to be wrong (3 minutes ahead). You can confirm this with sites like https://time.gov or https://uhr.ptb.de

To resolve this you could enable [time synchronization](https://wiki.archlinux.org/title/System_time#Time_synchronization).

**Reply by BlackDex** [reply]
The problem here is that you computer/browser is a few minutes off.
It reports an error if the time drift is 15 seconds or more.

So, if you fix your computers date/time, it will not report that error.
The server time compared to the NTP is ok, and thus has no issues.

**Reply by xkpx64** [reply]
Thanks both of you, Sync from windows settings fixed the problem ! <3 + karma & beer/steak
